### PR TITLE
rubyfmt: init at 0.8.1

### PR DESCRIPTION
**Pull request by bobvanderlinden**
###### Description of changes

This adds the rubyfmt tool; a formatter for Ruby files.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by AndersonTorres — APPROVED** [review]

**Comment by bobvanderlinden** [comment]
Thanks for the review :+1:

**Comment by AndersonTorres** [comment]
Ofborg is failing on many platforms.

File: pkgs/development/tools/rubyfmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    readline
+    libiconv
+    libobjc
+    libunwind
+    Foundation
+    Security
+  ];
+
+  preConfigure = ''
+    (cd librubyfmt/ruby_checkout && autoreconf --install --force --verbose && ./configure)
```
Comment (AndersonTorres) [inline pkgs/development/tools/rubyfmt/default.nix:51]:
```suggestion
    pushd librubyfmt/ruby_checkout 
    autoreconf --install --force --verbose 
    ./configure
    popd
```

File: pkgs/development/tools/rubyfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  preFixup = ''
+    mv $out/bin/rubyfmt{-main,}
+  '';
+
+  meta = with lib; {
+    description = "A Ruby autoformatter";
+    homepage = "https://github.com/fables-tales/rubyfmt";
+    license = licenses.mit;
+    platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/development/tools/rubyfmt/default.nix:73]:
```suggestion
```

File: pkgs/development/tools/rubyfmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rubyfmt";
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner = "fables-tales";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/tools/rubyfmt/default.nix:25]:
```suggestion
    repo = "rubyfmt";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by bobvanderlinden** [comment]
@ofborg build rustfmt

**Comment by bobvanderlinden** [comment]
@ofborg build rubyfmt

**Comment by bobvanderlinden** [comment]
Ugh 😅 mixed up the build command for ofborg.

Thanks @AndersonTorres for your feedback. I've incorporated your suggestions 👍

File: pkgs/development/tools/rubyfmt/default.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+, libobjc
+, libunwind
+, libxcrypt
+, Foundation
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rubyfmt";
+  version = "0.8.1";
```
Comment (kirillrdy) [inline pkgs/development/tools/rubyfmt/default.nix:21]: 0.9.5 is out, but no need to delay merging this

**Comment by kirillrdy** [comment]
:+1:

**Pull request merged**